ethereum/core/src/main/java/org/hyperledger/besu/ethereum/chain/ChainDataPruner.java

```diff
@@ -84,6 +87,12 @@ public void onBlockAdded(final BlockAddedEvent event) {
           }
           prunerStorage.setPruningMark(pruningTransaction, currentPruningMark);
           pruningTransaction.commit();
+          if (pruningRequired) {
+            // After pruning data, compact storage to reclaim space
+            if (blockchainStorage instanceof CompactableStorage compactable) {
+              compactable.compact();
+            }
+          }
```

Comment:
Totally agree with Simon here.  Forcing compaction because of periodic chain pruning is likely to have implications for block processing performance and IO generally.

I definitely want to see performance stats on this PR.  Charts (grafana) of IO usage with and without this feature, correlated with block execution time.